### PR TITLE
createBlog

### PR DESCRIPTION
**Pull request by pjmiles**
Created the first function.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,2 +1,3 @@
 PORT = 7000
-dbURI = mongodb+srv://newblog:newblog@cluster0.xtors.mongodb.net/?retryWrites=true&w=majority
\ No newline at end of file
+dbURI = mongodb+srv://newblog:newblog@cluster0.xtors.mongodb.net/?retryWrites=true&w=majority
```
Comment (iamr0b0tx) [inline .env:2]: ask @asylcreek

**Review by iamr0b0tx — CHANGES_REQUESTED** [review]

File: .env [hunk 2]
```diff
@@ -1,2 +1,3 @@
...
+DBURI = mongodb://localhost:27017/newblog
```
Comment (Asylcreek) [inline .env]: You should not commit your `.env` file.
It should be `NAME=VALUE`, not `NAME = VALUE`

Comment (Asylcreek) [reply .env]: You can use '_' to separate words

File: controller/createBlog.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const BlogPost = require("../model/blogSchema");
+
+const createBlogPost = async (req, res) => {
+  const { title, content, author } = req.body;
+  const blogPost = await new BlogPost({ title, content, author });
```
Comment (Asylcreek) [inline controller/createBlog.js:5]: You cannot `await` a class instantiation

File: controller/createBlog.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  blogPost.save();
```
Comment (Asylcreek) [inline controller/createBlog.js:6]: You are supposed to `await` this

Comment (Asylcreek) [reply controller/createBlog.js:6]: And read up on `Model.create` and use it instead

File: index.js [hunk 5]
```diff
@@ -4,18 +4,39 @@ const app = express();
...
 app.use(cors());
 app.use(express.urlencoded({ extended: true }));
 
+//Home page
+app.get("/", (req, res) => {
+    res.send("Welcome to PJMiles API");
+  });
+  
+// routes & API
+const Blog = require('./routes/blogRoutes');
```
Comment (Asylcreek) [inline index.js:18]: All your imports should be at the top of the file

File: index.js [hunk 6]
```diff
@@ -4,18 +4,39 @@ const app = express();
...
+
+
+
+
 
+// database 
 const database = () => {
     const db = require('./config/mongoose')
     return db
 }
```
Comment (Asylcreek) [inline index.js:37]: Why?

File: routes/blogRoutes.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+const router = require('express').Router();
+
+const { createBlogPost } = require('../controller/createBlog')
+
+
+router.route('/blogpost').post(createBlogPost)
```
Comment (Asylcreek) [inline routes/blogRoutes.js:6]: `.route('/')` instead

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (pjmiles) [reply .env:2]: ok

File: .env (hunk 2 above)
Comment (pjmiles) [reply .env]: ok

File: index.js (hunk 5 above)
Comment (pjmiles) [reply index.js:18]: ok

File: index.js (hunk 6 above)
Comment (pjmiles) [reply index.js:37]: I wanted to keep the database connection separate

File: routes/blogRoutes.js (hunk 7 above)
Comment (pjmiles) [reply routes/blogRoutes.js:6]: ok

File: index.js (hunk 6 above)
Comment (Asylcreek) [reply index.js:37]: Why?

**Pull request merged**